pytezos/operation/group.py

```diff
@@ -74,8 +75,28 @@ def _spawn(self, **kwargs):
             signature=kwargs.get('signature', self.signature)
         )
 
+    def _adjust_counter(self) -> None:
+        """Adjust counter based on pending transactions in mempool.
+        """
+        current_counter = self.context.counter
+        key_hash = self.key.public_key_hash()
+        mempool = self.shell.mempool.pending_operations()
+
+        for status, operations in mempool.items():
+            for operation in operations:
+                # FIXME: Replace with pattern matching based on operation kind
```

Comment:
Actually not by kind but mempool status (applied, branch_refused, branch_delayed, etc).
Anyways, from my POV this workaround with pattern matching is good enough